### PR TITLE
Add basic caching to CardSetApiClient

### PR DESCRIPTION
**Pull request by asimonov-im**

### CODE REVIEW

**Comment by asimonov-im** [comment]
Looks like Valve [broke their API](https://github.com/ValveSoftware/ArtifactDeckCode/issues/21) (hopefully temporarily. 😭

File: ArtifactDeckCodeDotNet/CardSetApiClient.cs [hunk 1]
```diff
@@ -1,65 +1,130 @@
...
         }
 
-        public async Task<CardSetData> GetCardSetAsync(string setId)
+        public async Task<CardSet> GetCardSetAsync(string setId, bool forceFetch = false)
         {
-            await GetJsonLocation(setId);
+            CardSetDataCache dataCache = _cache.Value;
+            if (forceFetch ||
+                !dataCache.TryGetValue(setId, out var data) ||
+                data.ExpireTimeUtc < DateTimeOffset.Now)
```
Comment (MisterJimson) [inline ArtifactDeckCodeDotNet/CardSetApiClient.cs]:
```suggestion
                data.ExpireTimeUtc < DateTimeOffset.UtcNow)
```

Should be UtcNow I think.

File: ArtifactDeckCodeDotNet.Tests/CardSetApiClientTests.cs [hunk 2]
```diff
@@ -1,20 +1,43 @@
...
-using Xunit;
+﻿using Xunit;
 
 namespace ArtifactDeckCodeDotNet.Tests
 {
     public class CardSetApiClientTests
     {
         [Fact]
-        public async Task GetCardSetAsyncShouldReturnCardSet()
+        public async void GetCardSetAsyncShouldReturnDataForSet00()
```
Comment (MisterJimson) [inline ArtifactDeckCodeDotNet.Tests/CardSetApiClientTests.cs]: I prefer async Task to void, even in tests.

File: ArtifactDeckCodeDotNet/CardSetApiClient.cs [hunk 3]
```diff
@@ -1,65 +1,130 @@
...
-    public class CardSetApiClient
+    public class CardSetApiClient : IDisposable
     {
         HttpClient _httpClient;
-        Dictionary<string, CardSetJsonLocation> _jsonLocations = new Dictionary<string, CardSetJsonLocation>();
-        Dictionary<string, CardSetData> _cardSetData = new Dictionary<string, CardSetData>();
+        Lazy<CardSetDataCache> _cache;
 
-        public CardSetApiClient()
+        internal static string DataCacheFile = Path.Combine(Path.GetTempPath(), "ArtifactSetDataCache.json");
```
Comment (MisterJimson) [inline ArtifactDeckCodeDotNet/CardSetApiClient.cs]:
```suggestion
        private static string DataCacheFile = Path.Combine(Path.GetTempPath(), "ArtifactSetDataCache.json");
```

**Review by MisterJimson — CHANGES_REQUESTED** [review]
Awesome, thanks for the help. Looking good, just a few minor comments.

File: ArtifactDeckCodeDotNet.Tests/CardSetApiClientTests.cs (hunk 2 above)
Comment (asimonov-im) [reply ArtifactDeckCodeDotNet.Tests/CardSetApiClientTests.cs]: Gotcha.

**Review by MisterJimson — APPROVED** [review]
You're the best!

**Pull request merged**